packages/databases-collections/src/components/collections.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useCallback } from 'react';
+import { connect } from 'react-redux';
+import { CollectionsList } from '@mongodb-js/databases-collections-list';
+import {
+  Banner,
+  BannerVariant,
+  css,
+  spacing,
+} from '@mongodb-js/compass-components';
+import { useTrackOnChange } from '@mongodb-js/compass-logging/provider';
+import {
+  refreshCollections,
+  type CollectionsState,
+  openCollection,
+  deleteCollection,
+  createNewCollection,
+} from '../modules/collections';
+import type Collection from 'mongodb-collection-model';
+import toNS from 'mongodb-ns';
+
+const ERROR_WARNING = 'An error occurred while loading collections';
+
+const collectionsErrorStyles = css({
+  padding: spacing[3],
+});
+
+type CollectionsListProps = {
+  namespace: string;
+  collections: Collection[];
+  collectionsLoadingStatus: string;
+  collectionsLoadingError?: string | null;
+  isEditable: boolean;
+  onCollectionClick(ns: string): void;
+  onDeleteCollectionClick(ns: string): void;
+  onCreateCollectionClick(dbName: string): void;
+  onRefreshClick(): void;
+};
+
+const Collections: React.FunctionComponent<CollectionsListProps> = ({
+  namespace,
+  collections,
+  collectionsLoadingStatus,
+  collectionsLoadingError,
+  isEditable,
+  onCollectionClick,
+  onDeleteCollectionClick,
+  onCreateCollectionClick: _onCreateCollectionClick,
+  onRefreshClick,
+}) => {
+  useTrackOnChange(
+    'COMPASS-COLLECTIONS-UI',
+    (track) => {
+      track('Screen', { name: 'collections' });
+    },
+    []
+  );
+
+  const onCreateCollectionClick = useCallback(() => {
+    _onCreateCollectionClick(toNS(namespace).database);
+  }, [namespace, _onCreateCollectionClick]);
+
+  if (collectionsLoadingStatus === 'error') {
+    return (
+      <div className={collectionsErrorStyles}>
+        <Banner variant={BannerVariant.Danger}>
+          {collectionsLoadingError
+            ? `${ERROR_WARNING}: ${collectionsLoadingError}`
+            : ERROR_WARNING}
+        </Banner>
+      </div>
+    );
+  }
+
+  const actions = Object.assign(
+    { onCollectionClick, onRefreshClick },
+    isEditable ? { onDeleteCollectionClick, onCreateCollectionClick } : {}
```

Comment:
Do we have a test for the readonly state not showing actions?
I don't think we cover it in e2e, it might be nice to add one if we don't.

Reply:
I have a functional test for this in the plugin itself, but for e2e we'd actually need to do some work first to allow for that, we don't test any special readonly behavior e2e at the moment and would need to at least have a user set up for that in the local db. I think there was a ticket for that, if not, I'll open one

Reply:
Sorry I didn't mean for e2e, was thinking unit/functional. Mis-wrote that message. Sounds good

Reply:
Ah, gotcha, In that case yeah, there is a test that updates topology on the instance and checks that the buttons went away